src/App.tsx

```diff
@@ -1,223 +1,293 @@
+  const [todosToUpdate, setTodosToUpdate] = useState<Todo[]>([]);
+
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
+    if (user) {
+      getTodos(user.id)
+        .then(setTodos)
+        .catch(() => {
+          setIsError(true);
+          setErrorMessage(ErrorTypes.UnableToLoad);
+        });
+    }
+
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
+  }, [user]);
+
+  const handleButtonClickAll = () => {
+    setFilterType(FilterTypes.All);
+  };
+
+  const handleButtonClickActive = () => {
+    setFilterType(FilterTypes.ACTIVE);
+  };
+
+  const handleButtonClickCompleted = () => {
+    setFilterType(FilterTypes.COMPLETED);
+  };
+
+  // const onSubmit = useCallback((event: React.FormEvent<HTMLFormElement>) => {
+  //   event.preventDefault();
+
+  //   if (!todoTitle.trim()) {
+  //     setErrorMessage(ErrorTypes.EmptyTitle);
+  //     setTodoTitle('');
+
+  //     return;
+  //   }
+
+  //   setIsLoading(true);
+
+  //   if (user) {
+  //     setTempTodo({
+  //       id: 0,
+  //       title: todoTitle,
+  //       completed: false,
+  //       userId: user.id,
+  //     });
+
+  //     addTodo(todoTitle, user.id)
+  //       .then(response => {
+  //         setTodos(prev => [...prev, {
+  //           id: response.id,
+  //           title: response.title,
+  //           completed: response.completed,
+  //           userId: response.userId,
+  //         }]);
+  //       })
+  //       .catch(() => setErrorMessage(ErrorTypes.UnableToAdd))
+  //       .finally(() => {
+  //         setIsLoading(false);
+  //         setTempTodo(null);
+  //         setTodoTitle('');
+  //       });
+  //   }
+  // }, [todoTitle, user]);
+
+  const addNewTodo = useCallback(async (newTitle: string) => {
+    if (!newTitle.trim()) {
+      setIsError(true);
+      setErrorMessage(ErrorTypes.EmptyTitle);
+
+      return;
+    }
+
+    setIsLoading(true);
+
+    if (user) {
+      try {
+        setTempTodo({
+          id: 0,
+          title: newTitle.trim(),
+          completed: false,
+          userId: user.id,
+        });
+
+        const newTodo = await addTodo({
+          title: newTitle.trim(),
+          userId: user?.id,
+          completed: false,
+        });
+
+        setTodos(prev => [...prev, newTodo]);
+        setErrorMessage('');
+        setIsError(false);
+      } catch (error) {
+        setIsError(true);
+        setErrorMessage(ErrorTypes.UnableToAdd);
+      } finally {
+        setIsLoading(false);
+        setTempTodo(null);
+      }
+    }
+  }, [user]);
+
+  const removeTodo = useCallback(async (todoId: number) => {
+    try {
+      setDeletingTodosIds(prev => [...prev, todoId]);
+
+      await deleteTodo(todoId);
+
+      setTodos(currentTodos => currentTodos.filter(todo => todo.id !== todoId));
+      setErrorMessage('');
+      setIsError(false);
```

Comment:
Let's create function showError to avoid code duplication, also this function could set timeout to close error message